astroplan/core.py

```diff
@@ -162,171 +207,851 @@ def altaz(self, time, target=None, obswl=None):
             return coordinate.transform_to(altaz_frame)
 
     # Sun-related methods.
+    @u.quantity_input(horizon=u.deg)
+    def _horiz_cross(self, t, alt, rise_set, horizon=0*u.degree):
+        '''
+        Find time ``t`` when values in array ``a`` go from
+        negative to positive or positive to negative (exclude endpoints)
 
-    def noon(self, time):
-        """
-        Returns the local, solar noon time.
+        ``return_limits`` will return nearest times to zero-crossing.
 
         Parameters
         ----------
-        time : `~astropy.time.Time`
-        """
-        raise NotImplementedError()
+        t : `~astropy.time.Time`
+            Grid of times
+        alt : `~astropy.units.Quantity`
+            Grid of altitudes
+        rise_set : {"rising",  "setting"}
+            Calculate either rising or setting across the horizon
+        horizon : float
+            Number of degrees above/below actual horizon to use
+            for calculating rise/set times (i.e.,
+            -6 deg horizon = civil twilight, etc.)
+        Returns
+        -------
+        Returns the lower and upper limits on the time and altitudes
+        of the horizon crossing.
+        '''
+        if rise_set == 'rising':
```

Comment:
Took me a minute to parse what was going on here -- maybe add a comment above the `if` that just says "# vectorized sign flip test or something"